Allow vars in remote packages (#4342)
OttoWinter/esphomedocs: components/packages.rst

        files: [file1.yml, file2.yml]  # optional; if not specified, all files will be included
        ref: main  # optional
        refresh: 1d  # optional
      
      remote_package_files2:
        url: https://github.com/esphome/non-existant-repo
        files:

>           - name: file1.yml

Reviewer: ```suggestion
          - path: file1.yml
```

            vars:
              a: 1
              b: 2

>           - name: file1.yml #Same file can be specified multiple times with different vars.

Reviewer: ```suggestion
          - path: file1.yml #Same file can be specified multiple times with different vars.
```

- **url** (**Required**, string): The URL for the repository.
- **username** (*Optional*, string): Username to be used for authentication, if required.
- **password** (*Optional*, string): Password to be used for authentication, if required.

> - **files** (**Required**, list of strings or list of { name, vars }): List of files to include.

Reviewer: ```suggestion
- **files** (**Required**, list of strings or list of ``{ name, vars }``): List of files to include.
```

> - **files** (**Required**, list of strings or list of ``{ name, vars }``): List of files to include.

Reviewer: ```suggestion
- **files** (**Required**): List of files to include. Can be one of:
  
  - list of file paths
  - list of objects containing ``path`` and ``vars``

```
